native/acter/src/api/client.rs

```diff
@@ -333,29 +334,37 @@ impl Client {
         self.room_by_alias_typed(&room_alias).await
     }
 
-    pub fn room_by_id_typed(&self, room_id: &OwnedRoomId) -> Option<Room> {
+    pub fn room_by_id_typed(&self, room_id: &RoomId) -> Option<Room> {
         self.core
             .client()
             .get_room(room_id)
             .map(|room| Room::new(self.core.clone(), room))
     }
 
-    pub async fn room_by_alias_typed(&self, room_alias: &OwnedRoomAliasId) -> Result<Room> {
-        for r in self.core.client().rooms() {
+    pub fn room_by_id(&self, room_id: &RoomId) -> Result<SdkRoom> {
+        self.core
+            .client()
+            .get_room(room_id)
+            .context("Room not found")
+    }
```

Comment:
okay this becomes very confusing in naming now ... we should at least put some doc comment on each of them then, no?

also the idea of `_typed` here was to separate `string`-based ffi-API from internally rust-typed API ... 

but possibly something for a future PR then... also with uniffi we might get rid of all of that finally...!

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1671/commits/7f4d5a8d4b9c3d0a33c890ac86f912adec53e44b

Reply:
Appended description to each `***_typed()` fn in https://github.com/acterglobal/a3/pull/1671/commits/291683c29afa91edda98414778cc4a6727d15c92